docs/en/setup/service-agent/java-agent/Application-toolkit-tracer.md

```diff
@@ -0,0 +1,213 @@
+
+  ```java
+  import Tracer;
+  ...
+    
+  SpanRef spanRef = Tracer.createEnteySpan("${operationName}", null);
+  ```
+
+* Use `Tracer.createLocalSpan()` API to create local span, the only parameter is the operation name of span.
+
+  ```java
+  import Tracer;
+  ...
+    
+  SpanRef spanRef = Tracer.createLocalSpan("${operationName}");
+  ```
+
+* Use `Tracer.createExitSpan()` API to create exit span
+
+  * **two parameters case**: the first parameter is the operation name of span, the second parameter is the remote peer which means the peer address of exit operation.
+
+    ```java
+    import Tracer;
+    ...
+      
+    SpanRef spanRef = Tracer.createExitSpan("${operationName}", "${remotePeer}");
+    ```
+
+  * **three parameters case**: the first parameter is the operation name of span, the second parameter is the `ContextCarrierRef` instance and the third parameter is the remote peer. This case will be introduced in **inject/extract** scenario.
+
+* Use `Tracer.stopSpan()` API to stop current span
+
+  ```java
+  import Tracer;
+  ...
+    
+  Tracer.stopSpan();
+  ```
+
+# Inject/extract
+
+The Inject/extract is to pass context information between different process. The `ContextCarrierRef` contains the reference of `ContextCarrier` and the `CarrierItemRef` contains the reference of CarrierItem. The `CarrierItem` instances compose a linked list. 
+
+* Use `Tracer.inject()` to inject information of current context into carrier
+* Use `Tracer.extract()` to extract info from contextCarrier. 
+* Use `items()` of `ContextCarrierRef` instance to get head `CarrierItemRef` instance.  
+* Use `hasNext()` of `CarrierItemRef` instance to judge if the `CarrierItemRef` has next item.
+* Use `next()` of `CarrierItemRef` instance to get next item
+* Use `getHeadKey` of `CarrierItemRef` instance to get key of current item
+* Use `getHeadValue` of `CarrierItemRef` instance to get value of current item
+* Use `setHeadValue` of `CarrierItemRef` instance to set value of current item
+
+```java
+/* 
+	You can consider map as the message body, such as Http, MQ and RPC. 
+	This code block do the inject operation in one process and then pass the map message.
+*/
+ContextCarrierRef contextCarrierRef = new ContextCarrierRef();
+Tracer.inject(contextCarrierRef);
+Map<String, String> map = new HashMap<>();
+CarrierItemRef next = contextCarrierRef.items();
+while (next.hasNext()) {
+    next = next.next();
+    map.put(next.getHeadKey(), next.getHeadValue());
+}
+...
+ 
+```
+
+```java
+// This code block receive the map message and do the extract operation in another process. 
```

Comment:
Why do we mention blocking?

Reply:
Because I want to remind users that these two code blocks are always used before sending the message and after receiving the message.

Reply:
So should I remove the "code block" pharse?

Reply:
Please remove, block usually means it is a time cost operation.
And notice map should be header/metadata, not the message itself.

Reply:
Ok, I will remove the "code block", and the "message" has been changed to "header/metadata".